src/main/java/bookbob/Main.java

```diff
@@ -40,12 +44,20 @@ public static void main(String[] args) throws IOException {
                 break;
 
             case "delete":
-                if (inputArr.length > 1) {
-                    String nric = inputArr[1].trim();
-                    commandHandler.delete(nric, records);
-                } else {
-                    System.out.println("Please specify an NRIC to delete.");
+                logger.log(Level.INFO, "Processing patient record deletion");
+                try {
+                    if (inputArr.length > 1) {
+                        String nric = inputArr[1].trim();
+                        commandHandler.delete(nric, records);
+                    } else {
+                        System.out.println("Please specify an NRIC to delete.");
+                        logger.log(Level.INFO, "Empty NRIC inputted.");
+                    }
+                } catch (Exception e) {
+                    logger.log(Level.WARNING, "Error processing deletion");
+                    System.out.println("Error in deleting files, specific error: " + e.getMessage());
```

Comment:
Good use of catch statement to catch an exception processing the delete command and print out the specific error message